### ISSUE TITLE
Matchers.scala ignores timestamp format specification and only uses the TypeMatcher

### ISSUE THREAD

**Issue by dberrueta-atlassian** [DiUS/pact-jvm#516]
In pact-jvm-matchers 2.x, a PACT like

```
          "$.body[0].write-timestamp": {
            "match": "timestamp",
            "timestamp": "yyyy-MM-dd\u0027T\u0027HH:mm:ss.SSS\u0027Z\u0027"
          }
```

erroneously rejects a value like `2017-08-11T02:54:42.522Z` with the error `Expected '2017-08-11T02:54:42.522Z' to be a timestamp`.

This is likely to be caused by `Matchers.scala` using the `typeMatcher` instead of the `timestampMatcher` to match timestamps. That causes to ignore the format specification, and only accept strict ISO formats.

**Comment by uglyog** [comment]
I'll backport the fix from 3.5.x

**Comment by dberrueta-atlassian** [comment]
Thank you. I was about to submit a PR with a 2.x-specific fix, but please backport the fix from 3.5.x. Would it be possible to build a release of 2.x with the bugfix?

**Comment by uglyog** [comment]
Yes, I can do that.

**Comment by uglyog** [comment]
Released 2.4.19 with all the fixes back-ported from 3.5.x

**Comment by dberrueta-atlassian** [comment]
I confirm that 2.4.19 fixes the bug.

Thank you for the quick response and release, I appreciate it!
